tck/features/MergeRelationshipAcceptance.feature

```diff
@@ -0,0 +1,833 @@
+      """
+      MATCH (b:B)
+      MERGE (a)-[r1:TYPE]->(b)<-[r2:TYPE]-(c)
+      RETURN type(r1), type(r2)
+      """
+    Then the result should be:
+      | type(r1) | type(r2) |
+      | 'TYPE'   | 'TYPE'   |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 2 |
+
+  Scenario: Using bound nodes in mid-pattern when pattern partly matches
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
+      CREATE (a)-[:TYPE]->(b)
+      """
+    When executing query:
+      """
+      MATCH (b:B)
+      MERGE (a:A)-[r1:TYPE]->(b)<-[r2:TYPE]-(c:C)
+      RETURN type(r1), type(r2)
+      """
+    Then the result should be:
+      | type(r1) | type(r2) |
+      | 'TYPE'   | 'TYPE'   |
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 2 |
+      | +labels        | 2 |
+
+  Scenario: Creating relationship using merged nodes
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (a:A), (b:B)
+      """
+    When executing query:
+      """
+      MERGE (a:A)
+      MERGE (b:B)
+      MERGE (a)-[:FOO]->(b)
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +relationships | 1 |
+
+  Scenario: Mixing `MERGE` with `CREATE`
+    Given an empty graph
+    When executing query:
+      """
+      CREATE (a:A)
+      MERGE (a)-[:KNOWS]->(b:B)
+      CREATE (b)-[:KNOWS]->(c:C)
+      RETURN count(*)
+      """
+    Then the result should be:
+      | count(*) |
+      | 1        |
+    And the side effects should be:
+      | +nodes         | 3 |
+      | +relationships | 2 |
+      | +labels        | 3 |
+
+  Scenario: Failing when creation would violate constraint
+    Given an empty graph
+    And having executed:
+      """
+      CREATE CONSTRAINT ON (p:Person) ASSERT p.id IS UNIQUE
+      """
+    And having executed:
+      """
+      CREATE (:Person {id: 666})
+      """
+    When executing query:
+      """
+      CREATE (a:A)
+      MERGE (a)-[:KNOWS]->(b:Person {id:666})
+      """
+    Then a ConstraintValidationFailed should be raised at runtime: CreateBlockedByConstraint
+
+  Scenario: Inside `FOREACH`, after match
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (s:S)
+      CREATE (s)-[:FOO]->({prop: 2})
+      """
+    When executing query:
+      """
+      MATCH (a:S)
+      FOREACH(x IN [1, 2, 3] |
+        MERGE (a)-[:FOO]->({prop: x})
+      )
+      """
+    Then the result should be empty
+    And the side effects should be:
+      | +nodes         | 2 |
+      | +relationships | 2 |
+      | +properties    | 2 |
+
+  Scenario: Inside `FOREACH`, after match, following other merge
+    Given an empty graph
+    And having executed:
+      """
+      CREATE (:S)
+      CREATE (:End {prop: 42})
+      """
+    When executing query:
+      """
+      MATCH (a:S)
+      FOREACH(x IN [42] |
+        MERGE (b:End {prop: x})
```

Comment:
This (indenting after `FOREACH`) is nice... should this be added to the style guide too?